bug: nil pointer dereference in agentsocket.Client.SyncReady when RPC fails

## Description

[`SyncReady`](https://github.com/coder/coder/blob/d4cd982/agent/agentsocket/client.go#L97-L103) panics when the underlying RPC call fails because it accesses `resp.Ready` without checking for errors first.

## Root Cause

```go
func (c *Client) SyncReady(ctx context.Context, unitName unit.ID) (bool, error) {
    resp, err := c.client.SyncReady(ctx, &proto.SyncReadyRequest{
        Unit: string(unitName),
    })
    return resp.Ready, err  // panics when err != nil (resp is nil)
}
```

Compare with [`SyncStatus`](https://github.com/coder/coder/blob/d4cd982/agent/agentsocket/client.go#L105-L130) in the same file, which correctly checks the error first.

## Stack Trace

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x15a9323]

goroutine 1 [running]:
github.com/coder/coder/v2/agent/agentsocket.(*Client).SyncReady(...)
    /home/runner/work/coder/coder/agent/agentsocket/client.go:102 +0x83
github.com/coder/coder/v2/cli.(*RootCmd).syncCommand.(*RootCmd).syncStart.func3(...)
    /home/runner/work/coder/coder/cli/sync_start.go:72 +0x4c5
```

## Suggested Fix

```go
func (c *Client) SyncReady(ctx context.Context, unitName unit.ID) (bool, error) {
    resp, err := c.client.SyncReady(ctx, &proto.SyncReadyRequest{
        Unit: string(unitName),
    })
    if err != nil {
        return false, err
    }
    return resp.Ready, nil
}
```

## Trigger

Observed during workspace startup when transient infrastructure issues (Kubernetes CNI/CSI initialization) caused the agent socket connection to fail. Any RPC failure will trigger this panic.

--- a/agent/agentsocket/client.go
+++ b/agent/agentsocket/client.go
@@ -99,7 +99,10 @@ func (c *Client) SyncReady(ctx context.Context, unitName unit.ID) (bool, error) 
 	resp, err := c.client.SyncReady(ctx, &proto.SyncReadyRequest{
 		Unit: string(unitName),
 	})
-	return resp.Ready, err
+	if err != nil {
+		return false, xerrors.Errorf("sync ready: %w", err)
+	}
+	return resp.Ready, nil
 }
 
 // SyncStatus gets the status of a unit and its dependencies.

